pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -112,15 +112,10 @@ type TresorCASpec struct {
 }
 
 // MeshRootCertificateStatus defines the status of the MeshRootCertificate resource
-type MeshRootCertificateStatus struct {
+type ProviderState struct {
```

Comment:
nit: Why change the name? I think MRCStatus is still more of an apt description than ProviderState. To me, the provider is the client, and its state would be something like up or down.

The MRC on the other hand, is really the status/state we're discussing.